xwayland/xwm.c

```diff
@@ -1165,6 +1177,19 @@ static void xwm_handle_net_wm_state_message(struct wlr_xwm *xwm,
 
 		wlr_signal_emit_safe(&xsurface->events.request_maximize, xsurface);
 	}
+
+	if (minimized != xsurface->minimized) {
+		if (xsurface->minimized) {
+			xsurface->saved_width = xsurface->width;
+			xsurface->saved_height = xsurface->height;
+		}
+
+		struct wlr_xwayland_minimize_event minimize_event;
```

Comment:
Can we use an initializer here to avoid future fields from being undefined?

```c
struct wlr_xwayland_minimize_event event = {
	.surface = xsurface,
	…
};
```